### PR TITLE
Cleanup - Removed unused imports and unused files

### PR DESCRIPTION
**Pull request by MounikaBattu17**
**Why does this PR accomplish?**
- Removed unused imports in all generated files.
- Deleted `read_functions.py` and `write_functions.py`
- Replaced interpreter call for `DAQmxGetChanType` in `channel.py`

**Why should this PR be merged?**
Implements the [Task 2383769](https://dev.azure.com/ni/DevCentral/_workitems/edit/2383769): Clean up - Delete unused files and imports and [Task 2383894](https://dev.azure.com/ni/DevCentral/_workitems/edit/2383894): Replace DAQmxGetChanType call in channel with library interpreter call

**What testing has been done?**
Automated running tests using `poetry run pytest`.

### CODE REVIEW

**Review by keer1111 — APPROVED** [review]

File: src/codegen/templates/system/device.py.mako [hunk 1]
```diff
@@ -9,16 +9,10 @@
...
-import numpy
 import deprecation
 
 from nidaqmx import utils
-from nidaqmx._lib import (
-    lib_importer, wrapped_ndpointer, enum_bitfield_to_list, ctypes_byte_str,
-    c_bool32)
-from nidaqmx.errors import (
-    check_for_error, is_string_buffer_too_small, is_array_buffer_too_small)
+from nidaqmx._lib import enum_bitfield_to_list
```
Comment (bkeryan) [inline src/codegen/templates/system/device.py.mako:15]: [outside the scope of this PR] At this point, we really shouldn't have to import `_lib` into any submodule other than `_library_interpreter` or `physical_channel`.

Here's where we're importing it:
```
PS C:\dev\nidaqmx-python> rg "_lib\b" src
src\handwritten\_lib.py
160:            self._import_lib()
166:            self._import_lib()
181:    def _import_lib(self):

src\handwritten\errors.py
166:    from nidaqmx._lib import lib_importer

src\codegen\templates\_library_interpreter.py.mako
13:from nidaqmx._lib import lib_importer, ctypes_byte_str, c_bool32, wrapped_ndpointer

src\codegen\templates\system\physical_channel.py.mako
19:from nidaqmx._lib import (

src\codegen\templates\system\device.py.mako
15:from nidaqmx._lib import enum_bitfield_to_list
```

I think we should do the following in a separate PR:
- Copy `check_for_errors`, `is_string_buffer_too_small`, and `is_array_buffer_too_small` from `errors.py` to `_library_interpreter.py`.
- Change the `errors.py` version of these functions into deprecated proxy functions:
```
@deprecation.deprecated(deprecated_in="0.8.0", details="This function will be removed in a future update.")
def check_for_error(error_code, samps_per_chan_written=None, samps_per_chan_read=None):
    import nidaqmx._library_interpreter
    return nidaqmx._library_interpreter.check_for_error(error_code, samps_per_chan_written, samps_per_chan_read)
```
- Move `enum_bitfield_to_list` and `enum_list_to_bitfield` to a private submodule. Perhaps a new `_bitfield_utils.py`?

File: src/handwritten/__init__.py [hunk 2]
```diff
@@ -1,7 +1,6 @@
 from nidaqmx.errors import DaqError, DaqReadError, DaqWriteError, DaqWarning, DaqResourceWarning
 from nidaqmx.scale import Scale
 from nidaqmx.task import Task
-from nidaqmx._task_modules.read_functions import CtrFreq, CtrTick, CtrTime
```
Comment (bkeryan) [inline src/handwritten/__init__.py:4]: These are public aliases for clients to use. Don't remove them. Instead, change them to import from the correct submodule, `nidaqmx.types`.

**Review by bkeryan — APPROVED** [review]
Approved with suggestions:
- Restore `CtrFreq` etc. in `__init__.py`, but import them from `nidaqmx.types`

File: src/codegen/templates/system/device.py.mako (hunk 1 above)
Comment (MounikaBattu17) [reply src/codegen/templates/system/device.py.mako:15]: Sure @bkeryan! This will be tracked in AB#2385784. Hence resolving this conversation.

**Pull request merged**